src/config/mod.rs

```diff
@@ -370,13 +371,14 @@ impl Config {
     }
 
     #[allow(dead_code)]
-    pub(super) fn from_toml(toml: &str, dir: &Path) -> Result<Config, String> {
-        Self::from_toml_for_style_edition(toml, dir, None, None, None)
+    pub(super) fn from_toml(toml: &str, dir: &Path,file_path : &Path) -> Result<Config, String> {
+        Self::from_toml_for_style_edition(toml, dir,file_path, None, None, None)
```

Comment:
We don't need to pass both `dir` and `file_path`. The `dir` can easily be derived from the file path.